### PR TITLE
Devex 1018: refactor advancedDocumentSearch

### PR DESCRIPTION
**Pull request by rosiehsmith**
- Extract methods from advancedDocumentSearch including: getSortQuery, getSealedQuery, getJudgeFilterForOrderSearch, getJudgeFilterForOpinionSearch and added unit tests
- Consolidate naming in advancedDocumentSearch to use document instead of docketEntry
- Use documentEventCodes for both opinionTypes AND order event codes which cleans up duplication
- Rename sortOrder to sortField, which is more accurate to what it's used for
- Move variable declarations closer to invocations
- Change judge filtering for order & opinion search to be in the documentQuery.filter instead of documentQuery.must since scoring shouldn't be needed for judge names.

### CODE REVIEW

File: shared/src/persistence/elasticsearch/advancedDocumentSearch.js [hunk 1]
```diff
@@ -1,12 +1,22 @@
-/* eslint-disable complexity */
```
Comment (TomElliottFlexion) [inline shared/src/persistence/elasticsearch/advancedDocumentSearch.js:1]: Yessssssss

**Review by TomElliottFlexion — APPROVED** [review]

File: shared/src/persistence/elasticsearch/advancedDocumentSearchHelpers/getJudgeFilterForOrderSearch.test.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  it('does a search for signed judge name', () => {
+    const result = getJudgeFilterForOrderSearch({
+      judgeName: 'Judge Alex Guarnaschelli',
+    });
+
+    expect(result).toEqual({
+      bool: {
+        should: {
+          match: {
+            'signedJudgeName.S': {
```
Comment (halprin) [inline shared/src/persistence/elasticsearch/advancedDocumentSearchHelpers/getJudgeFilterForOrderSearch.test.js]: To make this test less brittle, I would import `ORDER_JUDGE_FIELD` from `EntityConstants` and use it on this line.  This way, if `ORDER_JUDGE_FIELD` ever changes, this test won't fail.

File: shared/src/persistence/elasticsearch/advancedDocumentSearchHelpers/getJudgeFilterForOrderSearch.test.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+              operator: 'and',
+              query: 'Judge Alex Guarnaschelli',
```
Comment (halprin) [inline shared/src/persistence/elasticsearch/advancedDocumentSearchHelpers/getJudgeFilterForOrderSearch.test.js]: This is something I like to do, and is kind-of a stylistic opinion.
Can you add a `const judge = 'Judge Alex Guarnaschelli';` line on line "6.5", then use it in the name being passed in to `getJudgeFilterForOrderSearch` and in the expect here on the query line.

File: shared/src/persistence/elasticsearch/advancedDocumentSearchHelpers/getJudgeFilterForOpinionSearch.test.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+          {
+            match: {
+              'judge.S': 'Judge Antonia Lofaso',
+            },
+          },
+          {
+            match: {
+              'signedJudgeName.S': {
+                operator: 'and',
+                query: 'Judge Antonia Lofaso',
```
Comment (halprin) [inline shared/src/persistence/elasticsearch/advancedDocumentSearchHelpers/getJudgeFilterForOpinionSearch.test.js]: I made two comments in the below `getJudgeFilterForOrderSearch.test.js` test that could apply to this test too.

**Review by halprin — APPROVED** [review]
This is great!  The overall advance document search is a bit above me still, but splitting it up is awesome.  I did give a few comments for some suggestions.  Take them or leave them.

**Review by halprin — APPROVED** [review]
💥

**Pull request merged**